agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/AgentLogExporter.java

```diff
@@ -96,45 +99,62 @@ private CompletableResultCode internalExport(Collection<LogRecordData> logs) {
       return CompletableResultCode.ofFailure();
     }
     for (LogRecordData log : logs) {
-      try {
-        int severityNumber = log.getSeverity().getSeverityNumber();
-        if (severityNumber < severityThreshold) {
-          continue;
-        }
+      internalExport(log);
+    }
+    // always returning success, because all error handling is performed internally
+    return CompletableResultCode.ofSuccess();
+  }
 
-        String stack = log.getAttributes().get(ExceptionAttributes.EXCEPTION_STACKTRACE);
+  private void internalExport(LogRecordData log) {
+    try {
+      int severityNumber = log.getSeverity().getSeverityNumber();
+      if (severityNumber < severityThreshold) {
+        return;
+      }
 
-        SamplingOverrides samplingOverrides =
-            stack != null ? exceptionSamplingOverrides : logSamplingOverrides;
+      String stack = log.getAttributes().get(ExceptionAttributes.EXCEPTION_STACKTRACE);
 
-        SpanContext spanContext = log.getSpanContext();
+      SamplingOverrides samplingOverrides =
+          stack != null ? exceptionSamplingOverrides : logSamplingOverrides;
 
-        Double samplingPercentage = samplingOverrides.getOverridePercentage(log.getAttributes());
+      SpanContext spanContext = log.getSpanContext();
+      Double parentSpanSampleRate = log.getAttributes().get(AiSemanticAttributes.SAMPLE_RATE);
 
-        if (samplingPercentage != null && !shouldSample(spanContext, samplingPercentage)) {
-          continue;
-        }
+      AiSamplerForOverride sampler = samplingOverrides.getOverride(log.getAttributes());
 
-        if (samplingPercentage == null
-            && spanContext.isValid()
-            && !spanContext.getTraceFlags().isSampled()) {
-          // if there is no sampling override, and the log is part of an unsampled trace, then don't
-          // capture it
-          continue;
-        }
+      boolean hasSamplingOverride = sampler != null;
 
-        logger.debug("exporting log: {}", log);
+      if (!hasSamplingOverride
+          && spanContext.isValid()
+          && !spanContext.getTraceFlags().isSampled()) {
+        // if there is no sampling override, and the log is part of an unsampled trace,
```

Comment:
Because the use cases are not super simple, the code comments could try to refer to a test checking this scenario

Reply:
My suggestion was to refer to tests in the comments, in addition to the current comment explanations.

Reply:
I'd prefer to just improve the comments if they are unclear. Or if you make specific suggestions linking these to tests I'm happy to hit accept.